src/components/Login.js

```diff
@@ -30,18 +30,30 @@ class Login extends React.Component {
       let userData = {
         privateFields: {
           email: user.email,
-          createdOn: firebase.database.ServerValue.TIMESTAMP
+          createdOn: firebase.database.ServerValue.TIMESTAMP,
+          phoneNumber: '',
+          facebookId: '',
+          googleId: '',
+          twitterId: '',
+          githubId: '',
+          pushNotificationsToken: ''
         },
         publicFields: {
-          avatarImageUrl: user.photoURL || '',
-          displayName: user.displayName || user.email,
+          avatarImageUrl:
```

Comment:
Why is this made constant instead of `user.photoUrl`

Reply:
since in professor's latest rules, the avatarImageUrl must be a url and user.photoUrl is empty. Then the login will failed. I will use || to revise this